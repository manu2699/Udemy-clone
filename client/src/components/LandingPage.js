import React, { Component } from "react";

class LandingPage extends Component {
  render() {
    return (
      <div className="container">
        <h2 className="text-center">Udemy Clone</h2>
        <div className="jumbotron mt-5">
          <div className="col-sm-8 mx-auto">
            This is a udemy clone, made with react, node, express and mongo
          </div>
        </div>
      </div>
    );
  }
}

export default LandingPage;
