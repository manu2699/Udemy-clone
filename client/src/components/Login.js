import React, { Component } from "react";
import { login } from "./UserFunctions";
import validator from "validator";

class Login extends Component {
  state = {
    email: "",
    password: ""
  };

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();

    if (!validator.isEmail(this.state.email)) {
      alert("Enter correct email!");
      return;
    }

    if (validator.isEmpty(this.state.password)) {
      alert("Enter password!");
      return;
    }

    const user = {
      email: this.state.email,
      password: this.state.password
    };

    login(user).then(res => {
      if (!res.status) {
        this.props.history.push("/profile");
      } else {
        alert(res.message);
      }
    });
  }
  render() {
    return (
      <div className="container">
        <div className="row">
          <div className="col-md-6 mt-5 mx-auto">
            <form onValidate onSubmit={e => this.onSubmit(e)}>
              <h1 className="h2 mb-2 font-weight-bold">Login</h1>
              <div className="form-group">
                <label htmlFor="email">Email</label>
                <input
                  type="email"
                  className="form-control"
                  name="email"
                  placeholder="Enter Your Email"
                  value={this.state.email}
                  onChange={e => this.onChange(e)}
                />
                <label htmlFor="password">Password</label>
                <input
                  type="password"
                  className="form-control"
                  name="password"
                  placeholder="Enter Your Password"
                  value={this.state.password}
                  onChange={e => this.onChange(e)}
                />
              </div>
              <button
                className="btn btn-lg btn-primary btn-block"
                type="submit"
              >
                Login
              </button>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

export default Login;
