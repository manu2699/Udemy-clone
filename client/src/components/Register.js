import React, { Component } from "react";
import { register } from "./UserFunctions";
import validator from "validator";
import { withRouter } from "react-router";

class Register extends Component {
  state = {
    firstName: "",
    lastName: "",
    email: "",
    password: "",
    isTeacher: false
  };

  onChange(e) {
    if (e.target.name !== "isTeacher") {
      this.setState({ [e.target.name]: e.target.value });
    } else {
      this.setState({ [e.target.name]: e.target.checked });
    }
  }

  onSubmit(e) {
    e.preventDefault();

    if (!validator.isEmail(this.state.email)) {
      alert("please fill in the correct email!");
      return;
    }

    if (
      validator.isEmpty(this.state.firstName) &&
      validator.isEmpty(this.state.lastName) &&
      validator.isEmpty(this.state.password)
    ) {
      alert("please fill in all details!");
      return;
    }

    const user = {
      firstName: this.state.firstName,
      lastName: this.state.lastName,
      email: this.state.email,
      password: this.state.password,
      isTeacher: this.state.isTeacher
    };

    register(user).then(res => {
      if (res) {
        this.props.history.push("/login");
      }
    });
  }

  render() {
    return (
      <div className="container">
        <div className="row">
          <div className="col-md-6 mt-5 mx-auto">
            <form onValidate onSubmit={e => this.onSubmit(e)}>
              <h1 className="h2 mb-2 font-weight-bold">Register a Account</h1>
              <div className="form-group">
                <label htmlFor="firstName">First Name</label>
                <input
                  type="text"
                  className="form-control"
                  name="firstName"
                  placeholder="Your First Name"
                  value={this.state.firstName}
                  onChange={e => this.onChange(e)}
                />
                <label htmlFor="lastName">Last Name</label>
                <input
                  type="text"
                  className="form-control"
                  name="lastName"
                  placeholder="Your Last Name"
                  value={this.state.lastName}
                  onChange={e => this.onChange(e)}
                />
                <label htmlFor="email">Email</label>
                <input
                  type="email"
                  className="form-control"
                  name="email"
                  placeholder="Enter Your Email"
                  value={this.state.email}
                  onChange={e => this.onChange(e)}
                />
                <label htmlFor="password">Password</label>
                <input
                  type="password"
                  className="form-control"
                  name="password"
                  placeholder="Enter Your Password"
                  value={this.state.password}
                  onChange={e => this.onChange(e)}
                />
                <label htmlFor="isTeacher">Are you a Teacher?</label>
                <input
                  type="checkbox"
                  className="form-control"
                  name="isTeacher"
                  value={this.state.isTeacher}
                  onChange={e => this.onChange(e)}
                />
              </div>
              <button
                className="btn btn-lg btn-primary btn-block"
                type="submit"
              >
                Register
              </button>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

export default withRouter(Register);
