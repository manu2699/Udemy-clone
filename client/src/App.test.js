import React from "react";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

import LandingPage from "./components/LandingPage";
import Register from "./components/Register";
import Login from "./components/Login";

Enzyme.configure({ adapter: new Adapter() });

describe("<App />", () => {
  it("renders landing page", () => {
    const page = renderer.create(<LandingPage />).toJSON();
    expect(page).toMatchSnapshot();
  });

  it("renders register page", () => {
    const page = renderer.create(<Register />).toJSON();
    expect(page).toMatchSnapshot();
  });

  it("renders login page", () => {
    const page = renderer.create(<Login />).toJSON();
    expect(page).toMatchSnapshot();
  });
});
