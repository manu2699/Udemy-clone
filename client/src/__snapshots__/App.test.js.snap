// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<div
  className="container"
>
  <div
    className="jumbotron mt-5"
  >
    <div
      className="col-sm-8 mx-auto"
    >
      This is a udemy clone, made with react, node, express and mongo
    </div>
  </div>
</div>
`;

exports[`<App /> renders 2`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 mt-5 mx-auto"
    >
      <form
        onSubmit={[Function]}
        onValidate={true}
      >
        <h1
          className="h2 mb-2 font-weight-bold"
        >
          Register a Account
        </h1>
        <div
          className="form-group"
        >
          <label
            htmlFor="firstName"
          >
            First Name
          </label>
          <input
            className="form-control"
            name="firstName"
            onChange={[Function]}
            placeholder="Your First Name"
            type="text"
            value=""
          />
          <label
            htmlFor="lastName"
          >
            Last Name
          </label>
          <input
            className="form-control"
            name="lastName"
            onChange={[Function]}
            placeholder="Your Last Name"
            type="text"
            value=""
          />
          <label
            htmlFor="email"
          >
            Email
          </label>
          <input
            className="form-control"
            name="email"
            onChange={[Function]}
            placeholder="Enter Your Email"
            type="email"
            value=""
          />
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            placeholder="Enter Your Password"
            type="password"
            value=""
          />
          <label
            htmlFor="isTeacher"
          >
            Are you a Teacher?
          </label>
          <input
            className="form-control"
            name="isTeacher"
            onChange={[Function]}
            type="checkbox"
            value={false}
          />
        </div>
        <button
          className="btn btn-lg btn-primary btn-block"
          type="submit"
        >
          Register
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`<App /> renders landing page 1`] = `
<div
  className="container"
>
  <div
    className="jumbotron mt-5"
  >
    <div
      className="col-sm-8 mx-auto"
    >
      This is a udemy clone, made with react, node, express and mongo
    </div>
  </div>
</div>
`;

exports[`<App /> renders login page 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 mt-5 mx-auto"
    >
      <form
        onSubmit={[Function]}
        onValidate={true}
      >
        <h1
          className="h2 mb-2 font-weight-bold"
        >
          Login
        </h1>
        <div
          className="form-group"
        >
          <label
            htmlFor="email"
          >
            Email
          </label>
          <input
            className="form-control"
            name="email"
            onChange={[Function]}
            placeholder="Enter Your Email"
            type="email"
            value=""
          />
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            placeholder="Enter Your Password"
            type="password"
            value=""
          />
        </div>
        <button
          className="btn btn-lg btn-primary btn-block"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`<App /> renders register page 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 mt-5 mx-auto"
    >
      <form
        onSubmit={[Function]}
        onValidate={true}
      >
        <h1
          className="h2 mb-2 font-weight-bold"
        >
          Register a Account
        </h1>
        <div
          className="form-group"
        >
          <label
            htmlFor="firstName"
          >
            First Name
          </label>
          <input
            className="form-control"
            name="firstName"
            onChange={[Function]}
            placeholder="Your First Name"
            type="text"
            value=""
          />
          <label
            htmlFor="lastName"
          >
            Last Name
          </label>
          <input
            className="form-control"
            name="lastName"
            onChange={[Function]}
            placeholder="Your Last Name"
            type="text"
            value=""
          />
          <label
            htmlFor="email"
          >
            Email
          </label>
          <input
            className="form-control"
            name="email"
            onChange={[Function]}
            placeholder="Enter Your Email"
            type="email"
            value=""
          />
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            placeholder="Enter Your Password"
            type="password"
            value=""
          />
          <label
            htmlFor="isTeacher"
          >
            Are you a Teacher?
          </label>
          <input
            className="form-control"
            name="isTeacher"
            onChange={[Function]}
            type="checkbox"
            value={false}
          />
        </div>
        <button
          className="btn btn-lg btn-primary btn-block"
          type="submit"
        >
          Register
        </button>
      </form>
    </div>
  </div>
</div>
`;
